src/sentry/api/endpoints/project_details.py

```diff
@@ -512,3 +519,55 @@ def delete(self, request, project):
             )
 
         return Response(status=204)
+
+    def transfer_project(self, request, project):
+        email = request.DATA.get('transfer')
+
+        if email is None:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        try:
+            owner = OrganizationMember.objects.filter(
+                user__email__iexact=email,
+                role=roles.get_top_dog().id,
+                user__is_active=True,
+            )[0]
+        except IndexError:
+            return Response({'details': 'Could not find owner with that email'},
+                            status=status.HTTP_404_NOT_FOUND)
+
+        transaction_id = uuid4().hex
+        url_data = sign(
+            actor_id=request.user.id,
```

Comment:
i think you should probably check to make sure that `request.user.is_authenticated()` is true before doing this since I believe this will be an `AnonymousUser` when the auth method is an api key

Reply:
Should we only allow this for `is_authenticated` users then?